### PR TITLE
EBF-265: Add a `<Breadcrumb>` component

### PR DESCRIPTION
**Pull request by obyknovenius**

### CODE REVIEW

**Comment by obyknovenius** [comment]
The current implementation has the following issues.
1. Every nested `<a>` element must have `data-text` attribute to prevent the "jumping" effect on hover.
2. When using a static text as a segment, it must be manually set as `@active`.
3. When using the `<LinkTo>` component, Ember automatically adds the `active` attribute, which makes the corresponding segment look like it is the current page.

**Comment by moysa** [comment]
First off, I would suggest using the name `Breadcrumb`, I don't think that the `bar` suffix is necessary.

Concerning the three issues, I would use `LinkTo` as the default item component. You can set the `active` attribute of the `LinkTo` component to control it's state. So I would suggest the following:
```
<Breadcrumb as |bar|>
  <bar.item @route="docs" @label="Docs" />
  <bar.item @route="docs.breadcrumbs" @label="Breadcrumbs" @active={{true}}/>
</Breadcrumb>
```
and the item would render the `LinkTo` component. If an item has a block, then render what ever is in the block. This way we will have enough control while allowing for flexibility.

File: tests/dummy/app/components/docs/breadcrumb/index.hbs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<DocsDemo as |demo|>
+  <demo.example @name="breadcrumb.hbs">
+    <Breadcrumb as |bar|>
+      <bar.item @route="docs" @label="Home" />
+      <bar.item @route="docs.breadcrumbs" @label="Library" />
+      <bar.item @active={{true}}>Data</bar.item>
```
Comment (moysa) [inline tests/dummy/app/components/docs/breadcrumb/index.hbs:6]: Set this example as an active LinkTo. ie: `<bar.item @route="docs.breadcrumbs" @label="Docs" @active={{true}} />`

Example of block content can be in a separate Breadcrumb.

**Review by moysa — CHANGES_REQUESTED** [review]

File: tests/dummy/app/components/docs/breadcrumb/index.hbs (hunk 1 above)
Comment (obyknovenius) [reply tests/dummy/app/components/docs/breadcrumb/index.hbs:6]: Like this?
```hbs
<Breadcrumb as |bar|>
  <bar.item @route="docs" @label="Docs" />
  <bar.item @route="docs.breadcrumbs" @label="Breadcrumbs" @active={{true}} />
</Breadcrumb>
```

```hbs
<Breadcrumb as |bar|>
  <bar.item>
    <a href="#" data-text="Home">Home</a>
  </bar.item>
  <bar.item>
    <a href="#" data-text="Library">Library</a>
  </bar.item>
  <bar.item @active={{true}}>
    Data
  </bar.item>
</Breadcrumb>
```

Comment (moysa) [reply tests/dummy/app/components/docs/breadcrumb/index.hbs:6]: Yes

**Review by moysa — APPROVED** [review]

**Review by moysa — APPROVED** [review]

**Pull request merged**